### PR TITLE
t.py: init at 2020-04-11

### PR DESCRIPTION
**Pull request by illiusdope**
###### Motivation for this change

It's a nice, simple todo tool.  It's called `t` upstream, but since there's already a `t` in nixpkgs I've called it `t.py`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/t_py/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ lib, python3Packages, fetchFromGitHub }:
+
+python3Packages.buildPythonApplication {
+  pname = "t.py";
+  version = "2020-04-11";
```
Comment (sternenseemann) [inline pkgs/applications/misc/t_py/default.nix:5]:
```suggestion
  version = "unstable-2020-04-11";
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Otherwise lgtm. You can also probably use `pname = "t";` attribute name and package name don't always match in nixpkgs.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7307,6 +7307,8 @@ in
 
   t = callPackage ../tools/misc/t { };
 
+  t_py = callPackage ../applications/misc/t_py { };
```
Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix:7310]:
```suggestion
  t_py = python3Packages.callPackage ../applications/misc/t_py { };
```

File: pkgs/applications/misc/t_py/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{ lib, python3Packages, fetchFromGitHub }:
```
Comment (sternenseemann) [inline pkgs/applications/misc/t_py/default.nix:1]:
```suggestion
{ lib, buildPythonApplication, fetchFromGitHub }:
```

File: pkgs/applications/misc/t_py/default.nix [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+python3Packages.buildPythonApplication {
```
Comment (sternenseemann) [inline pkgs/applications/misc/t_py/default.nix:3]:
```suggestion
buildPythonApplication {
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Just pass `buildPythonApplication` from `python3Packages` instead of the whole package set. This doesn't make a huge difference, but is more in line with other python applications in top level nixpkgs.

**Comment by sternenseemann** [comment]
Result of `nixpkgs-review pr 98474` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- t_py
</details>

**Comment by sternenseemann** [comment]
`t` seems to have [tests](https://github.com/sjl/t/tree/master/tests). I'm not exactly sure how to run them, maybe you can find that out and add a `checkPhase`.

**Comment by sternenseemann** [comment]
Seems like steve is using cram: <https://github.com/brodie/cram>

**Comment by illiusdope** [comment]
I initially didn't bother with the tests because I didn't know what I was looking at, but that was super easy :shrug:

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build t_py

**Comment by sternenseemann** [comment]
Last (hopefully) order of business would be squashing all the commits into a single ` t_py: init at 2020-04-11`, I guess :)

**Review by sternenseemann — APPROVED** [review]

**Comment by sternenseemann** [comment]
Please rebase against `master` instead of using `git merge` to avoid unecessary merge commits when this PR is merged.

**Comment by Artturin** [comment]
Reopen and rebase if you're still interested in this.

**Pull request closed without merging**